src/linux/component.rs

```diff
@@ -1,160 +1,388 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
+// Information about values readable from `hwmon` sysfs.
+//
+// Values in /sys/class/hwmonN are `c_long` or `c_ulong`
+// transposed to rust we only read `u32` or `i32` values.
+//
 use crate::ComponentExt;
 
 use std::collections::HashMap;
-use std::fs::{metadata, read_dir, File};
+use std::fs::{read_dir, File};
 use std::io::Read;
 use std::path::{Path, PathBuf};
 
 #[doc = include_str!("../../md_doc/component.md")]
 pub struct Component {
-    temperature: f32,
-    max: f32,
-    critical: Option<f32>,
+    /// Optional Associated device of a Component
```

Comment:
Lots of capital letters in here. :laughing: 

What about:

```suggestion
    /// Optional associated device of a `Component`.
```

?